tests/end-to-end/upgrade/pkg/tests/eventmesh-migration/helpers/application.go

```diff
@@ -0,0 +1,145 @@
+	appbroker "github.com/kyma-project/kyma/components/application-broker/pkg/client/clientset/versioned"
+	appconnectorv1alpha1 "github.com/kyma-project/kyma/components/application-operator/pkg/apis/applicationconnector/v1alpha1"
+	appconnector "github.com/kyma-project/kyma/components/application-operator/pkg/client/clientset/versioned"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	messagingclientv1alpha1 "knative.dev/eventing/pkg/client/clientset/versioned/typed/messaging/v1alpha1"
+	serving "knative.dev/serving/pkg/client/clientset/versioned"
+)
+
+const (
+	integrationNamespace = "kyma-integration"
+)
+
+type ApplicationOption func(*appconnectorv1alpha1.Application)
+
+func CreateApplication(appConnectorInterface appconnector.Interface, name string, applicationOptions ...ApplicationOption) error {
+	application := &appconnectorv1alpha1.Application{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Application",
+			APIVersion: "applicationconnector.kyma-project.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: appconnectorv1alpha1.ApplicationSpec{
+			SkipInstallation: false,
+			Services:         []appconnectorv1alpha1.Service{},
+		},
+	}
+	for _, option := range applicationOptions {
+		option(application)
+	}
+	_, err := appConnectorInterface.ApplicationconnectorV1alpha1().Applications().Create(application)
+	return err
+}
+
+func WithAccessLabel(label string) ApplicationOption {
+	return func(application *appconnectorv1alpha1.Application) {
+		application.Spec.AccessLabel = label
+	}
+}
+
+func WithEventService(id string) ApplicationOption {
+	return func(application *appconnectorv1alpha1.Application) {
+		application.Spec.Services = append(application.Spec.Services,
+			appconnectorv1alpha1.Service{
+				ID:          id,
+				Name:        id,
+				DisplayName: "Some Events",
+				Description: "Application Service Class with Events",
+				Labels: map[string]string{
+					"connected-app": "app-name",
+				},
+				ProviderDisplayName: "provider",
+				Entries: []appconnectorv1alpha1.Entry{
+					{
+						Type: "Events",
+					},
+				},
+			})
+	}
+}
+
+func WaitForApplication(appConnector appconnector.Interface, messaging messagingclientv1alpha1.MessagingV1alpha1Interface, serving serving.Interface, name string, retryOptions ...retry.Option) error {
+	application, err := appConnector.ApplicationconnectorV1alpha1().Applications().Get(name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot get application: %+v", err)
+	}
+	if !application.Spec.SkipInstallation {
+		if err := WaitForChannel(messaging, name, integrationNamespace, retryOptions...); err != nil {
+			return fmt.Errorf("waiting for application failed: %+v, application: %+v", err, application)
+		}
+		if err := WaitForHttpSource(serving, name, integrationNamespace, retryOptions...); err != nil {
+			return fmt.Errorf("waiting for application failed: %+v, application: %+v", err, application)
```

Comment:
I would expect the error to contain all the information we need. Why dumping the entire object to the logs on top of that? It makes Prow logs hard to read.

Reply:
Cause it is important to understand the state of the object `Application` during a failure as rerunning is expensive.